components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -4286,6 +4338,77 @@ private PreparedStatement createConnectedAppsSqlStatement(Connection connection,
         return prepStmt;
     }
 
+    private PreparedStatement createConnectedAppsOfLocalAuthenticatorSqlStatement(Connection connection, String name,
+                                                                                  int tenantId, int limit, int offset)
+            throws SQLException, IdentityProviderManagementServerException {
+
+        String sqlQuery;
+        PreparedStatement prepStmt;
+        String databaseProductName = connection.getMetaData().getDatabaseProductName();
+        if (databaseProductName.contains("MySQL")
+                || databaseProductName.contains("MariaDB")
+                || databaseProductName.contains("H2")) {
+            sqlQuery = IdPManagementConstants.SQLQueries.GET_CONNECTED_APPS_LOCAL_MYSQL;
+            prepStmt = connection.prepareStatement(sqlQuery);
+            prepStmt.setString(1, name);
+            prepStmt.setInt(2, tenantId);
+            prepStmt.setInt(3, tenantId);
+            prepStmt.setString(4, name);
```

Comment:
As per the query, it expects idp name. But here we have passed the authenticator name.

Reply:
We store those authenticator name in the DB as IDP name. So by changing the authenticator name to `identityProviderName1` with https://github.com/wso2/carbon-identity-framework/pull/4518#discussion_r1171149710 will work?

Reply:
Or else let's add a comment explaining it.